src/shrpx_config.cc

```diff
@@ -242,8 +242,18 @@ read_tls_ticket_key_file(const std::vector<std::string> &files) {
 }
 
 FILE *open_file_for_write(const char *filename) {
-  auto fd = open(filename, O_WRONLY | O_CLOEXEC | O_CREAT | O_TRUNC,
-                 S_IRUSR | S_IWUSR);
+  #if defined O_CLOEXEC
+    auto fd = open(filename, O_WRONLY | O_CLOEXEC | O_CREAT | O_TRUNC,
+                     S_IRUSR | S_IWUSR);
+ #else
+    auto fd = open(filename, O_WRONLY | O_CREAT | O_TRUNC,
+                   S_IRUSR | S_IWUSR);
+ 
+    // We get race condition if execve is called at the same time.
+    if(fd != -1) {
+        int make_socket_closeonexec_(fd);
```

Comment:
This is not correct to me.
I proposed specifically `util::make_socket_closeonexec(fd);`